### PR TITLE
STY: add code check for use of builtin filter function

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref https://github.com/pandas-dev/pandas/pull/35717#issuecomment-674051885

### CODE REVIEW

File: ci/code_checks.sh [hunk 1]
```diff
@@ -179,6 +179,10 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     invgrep -R --include="*.py" -E "super\(\w*, (self|cls)\)" pandas
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
+    MSG='Check for use of builtin filter function' ; echo $MSG
+    invgrep -R --include="*.py" -P '(?<!def)[\(\s]filter\(' pandas
```
Comment (WillAyd) [inline ci/code_checks.sh:183]:
```suggestion
    invgrep -R --include="*.py" -P '(?<!def)filter\(' pandas
```

Is this necessary?

**Review by WillAyd — COMMENTED** [review]
looks good. potentially could simplify the regex otherwise @jreback

File: ci/code_checks.sh (hunk 1 above)
Comment (simonjayhawkins) [reply ci/code_checks.sh:183]: that would fail on uses of `.filter`, `simplefilter`, `.apply_filter` and more

**Review by WillAyd — APPROVED** [review]

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**